File: src/explore/javascript/explore.js
const fs = require('fs');
const Twilio = require('twilio');

/* Load client secrets from a local file: */
fs.readFile('../credentials.json', (err, content) => {
    if (err) return console.log("Error loading client secret file:\n", err);
    explore(JSON.parse(content));
});

function explore(credentials) {
    
    const { TWILIO_ACCOUNT_SID, TWILIO_AUTH_TOKEN } = credentials;
    const client = new Twilio(TWILIO_ACCOUNT_SID, TWILIO_AUTH_TOKEN);

    client.messages
        .list()
        .then(messages =>
            console.log(`The most recent message is:\n${messages[0].body}`)
        )
        .catch(err =>
            console.log(err)
        );

    console.log("Gathering your message log...")

}

File: src/explore/python/explore.py
import json
from twilio.rest import Client

f = open('../credentials.json')
credentials = json.load(f)

TWILIO_ACCOUNT_SID = credentials['TWILIO_ACCOUNT_SID']
TWILIO_AUTH_TOKEN = credentials['TWILIO_AUTH_TOKEN']
client = Client(TWILIO_ACCOUNT_SID, TWILIO_AUTH_TOKEN)

print("Message log:")
for msg in client.messages.list():
    print(msg.body)

MY_PHONE_NUMBER = credentials['MY_PHONE_NUMBER']
MY_TWILIO_PHONE_NUMBER = credentials['MY_TWILIO_PHONE_NUMBER']
new_msg = client.messages.create(
    to=MY_PHONE_NUMBER,
    from_=MY_TWILIO_PHONE_NUMBER,
    body="Coucou! (sent from Python)"
)
print(f"Just created a new message {new_msg.sid}:\n{new_msg.body}")
